pandasai/smart_datalake/__init__.py

```diff
@@ -351,144 +337,12 @@ def chat(self, query: str, output_type: Optional[str] = None):
-                    self._cache.set(self._get_cache_key(), code)
-
-            if self._config.callback is not None:
-                self._config.callback.on_code(code)
-
-            self.last_code_generated = code
-            self.logger.log(
-                f"""Code generated:
-```
-{code}
-```
-"""
-            )
-
-            retry_count = 0
-            code_to_run = code
-            result = None
-            while retry_count < self._config.max_retries:
-                try:
-                    # Execute the code
-                    context = CodeExecutionContext(self._last_prompt_id, self._skills)
-                    result = self._code_manager.execute_code(
-                        code=code_to_run,
-                        context=context,
-                    )
-
-                    break
-
-                except Exception as e:
-                    if (
-                        not self._config.use_error_correction_framework
-                        or retry_count >= self._config.max_retries - 1
-                    ):
-                        raise e
-
-                    retry_count += 1
-
-                    self._logger.log(
-                        f"Failed to execute code with a correction framework "
-                        f"[retry number: {retry_count}]",
-                        level=logging.WARNING,
-                    )
-
-                    traceback_error = traceback.format_exc()
-                    [
-                        code_to_run,
-                        reasoning,
-                        answer,
-                    ] = self._query_exec_tracker.execute_func(
-                        self._retry_run_code, code, traceback_error
-                    )
-
-            if result is not None:
-                if isinstance(result, dict):
-                    validation_ok, validation_logs = output_type_helper.validate(result)
-                    if not validation_ok:
-                        self.logger.log(
-                            "\n".join(validation_logs), level=logging.WARNING
-                        )
-                        self._query_exec_tracker.add_step(
-                            {
-                                "type": "Validating Output",
-                                "success": False,
-                                "message": "Output Validation Failed",
-                            }
-                        )
-                    else:
-                        self._query_exec_tracker.add_step(
-                            {
-                                "type": "Validating Output",
-                                "success": True,
-                                "message": "Output Validation Successful",
-                            }
-                        )
-
-                self.last_result = result
-                self.logger.log(f"Answer: {result}")
+            result = GenerateSmartDatalakePipeline(pipeline_context, self.logger).run()
```

Comment:
@milind-sinaptik  i would say why not pass query and output_type as dict to the pipeline.run(input). This way we can reduce the use of add_intermediate_value value usage.